### PR TITLE
Fix some errors (all reported by cppcheck)

### PR DESCRIPTION
**Pull request by stweil**
This is a first bunch of fixes for some real errors which were reported by the static code analyser cppcheck.

### CODE REVIEW

File: cf-agent/verify_networks.c [hunk 1]
```diff
@@ -415,7 +415,7 @@ static int GetArpInfo(ARPState **list, const Promise *pp)
     size_t line_size = CF_BUFSIZE;
     char *line = xmalloc(line_size);
     char comm[CF_BUFSIZE];
-    char ip[CF_MAX_IP_LEN], mac[CF_MAX_IP_LEN], device[CF_SMALLBUF], state[CF_SMALLBUF];
+    char ip[CF_MAX_IP_LEN + 1], mac[CF_MAX_IP_LEN], device[CF_SMALLBUF], state[CF_SMALLBUF];
```
Comment (kacfengine) [inline cf-agent/verify_networks.c]: Why not ``mac`` as well?

Comment (stweil) [inline cf-agent/verify_networks.c]: mac is filled with a maximum of 31 characters plus terminating \0, so only up to 32 bytes from the 64 byte buffer are used: the declaration allocates more memory than needed (that does not harm much).

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
Thanks, nice fixes!

**Comment by kacfengine** [comment]
Cherry-picked to 3.6.x as well.

**Pull request merged**